stan/math/opencl/matrix_cl.hpp

```diff
@@ -47,6 +47,8 @@ class matrix_cl<T, enable_if_arithmetic<T>> {
   // Forward declare the methods that work in place on the matrix
   template <matrix_cl_view matrix_view = matrix_cl_view::Entire>
   void zeros();
+  template <matrix_cl_view matrix_view = matrix_cl_view::Entire>
```

Comment:
Is `zeros()` supposed to be deleted on line 49?

Reply:
Also why the need for a strict zeros function?

Reply:
The thing is the current zeros kernel was working with lower and upper view already but in a "strict" fashion, always excluding the diagonal (Entire worked normally). 

In all places we were using zeros with Lower or Upper this was the expected behavior and the zeros kernel tests were also expecting the strict triangular behavior. So this was the intended behavior, I just found it odd that this was done this way without explicitly stating it. This one was on me.

I contemplated adding StrictlyUpper and StrictlyLower views, but that would mean we need to add the "strictly" behavior to all kernels that use the view. I dont think that is worth it.

So zeros<Lower>() on a matrix of ones results in

0 1 1
0 0 1
0 0 0 

and zeros_strict_tri<Lower() results in

1 1 1
0 1 1
0 0 1 

Hopefully I am making sense, I am half-way through my morning coffee :)

Reply:
Word!

Reply:
You deleted the code for this didnt you? 

```
template <matrix_cl_view matrix_view = matrix_cl_view::Entire> void zeros
```

How bad would it be to add the checks for strictly upper and strictly lower? And just have one zeros function? It feels like that would he nicer at the C++ level relative to having two methods

Reply:
No, the regular zeros remains, I did not remove it, there is also tests for it. 

To have a single zeros method we need to add `StrictlyUpper` and `StrictlyLower` to the view enum. That part is easy, extending the kernel would also be trivial.

Then its a question if we want to refactor all the other kernels to also support that behavior everywhere. Supporting those 2 enums only in zeros would be weired than this imo. I think a refactor to support `strictlyuppper`/`strictlylower` should be a separate issue/PR because that touches almost every kernel..

Reply:
The only other thing we could do is add a matrix_cl_strict_view enum and add another argument to the kernel. Would you prefer that? I am fine with that.

Reply:
Okay cool cool, if it's too much of a bother then I'm fine with a different method

Reply:
Adding another argument to denote if the view is strict or not is fine with me. Would you prefer that?